File: lessons/126/functions/python/function.py
import os
import datetime
import random
import boto3

bucket = os.getenv('BUCKET_NAME')
key = 'thumbnail.png'
table = 'Meta'

s3 = boto3.client('s3')
dynamodb = boto3.resource('dynamodb')


# Download the S3 object and return last modified date
def get_s3_object(bucket, key):
    object = s3.get_object(Bucket=bucket, Key=key)
    object_content = object['Body'].read()
    print(object_content)
    return object['LastModified']


# Save the item to the DynamoDB table
def save(table, date):
    table = dynamodb.Table(table)
    table.put_item(Item={'LastModified': date.isoformat()})


# Lambda handler
def lambda_handler(event, context):
    date = get_s3_object(bucket, key)
    random_number_of_days = random.randint(0, 100000)
    date += datetime.timedelta(days=random_number_of_days)
    save(table, date)

File: lessons/126/functions/nodejs/function.js
const aws = require('aws-sdk');
const s3 = new aws.S3();
const ddb = new aws.DynamoDB({ apiVersion: '2012-08-10' });

// Download the S3 object and print content
const getS3Object = (bucket, key, callback) => {
    const params = {
        Bucket: bucket,
        Key: key
    };

    s3.getObject(params, (err, data) => {
        if (err) return err;
        console.log(data);
        callback(data.LastModified);
    });
};

// Save the item to the DynamoDB table
const save = (table, date, callback) => {
    const modifiedDate = date.toISOString()
    const params = {
        TableName: table,
        Item: { 'LastModified': { S: modifiedDate } }
    };

    ddb.putItem(params, (err, data) => {
        if (err) return err;
        callback(null, 200);
    });
};

// Returns a random number between min and max
const getRandomInt = (min, max) => {
    return Math.random() * (max - min) + min;
};

exports.lambda_handler = (event, context, callback) => {
    const bucket = process.env.BUCKET_NAME;
    const key = 'thumbnail.png';
    const table = 'Meta';

    getS3Object(bucket, key, (date) => {
        const randomNumberOfDays = getRandomInt(0, 100000);
        date.setDate(date.getDate() + randomNumberOfDays);
        save(table, date, callback);
    });
};